src/confidential.rs

```diff
@@ -850,8 +849,84 @@ impl str::FromStr for AssetBlindingFactor {
     }
 }
 
+#[cfg(feature = "serde")]
+impl Serialize for AssetBlindingFactor {
+    fn serialize<S: Serializer>(&self, s: S) -> Result<S::Ok, S::Error> {
+        use bitcoin::hashes::hex::ToHex;
+        if s.is_human_readable() {
+            s.serialize_str(&self.to_hex())
```

Comment:
This is usually done using `collect_str` to avoid the allocation. `collect_str` takes a `fmt::Display` argument and uses that trait.

Reply:
Done, removed `use bitcoin::hashes::hex::ToHex` now unused, added a couple of lines in the added test to test serialization as well.